shiny/session/_session.py

```diff
@@ -138,6 +138,73 @@ def __instancecheck__(self, __instance: Any) -> bool:
         return isinstance(__instance, SessionProxy)
 
 
+class OutputInfo:
+    _input: Inputs
+    _id: Id
+
+    def __init__(self, input: Inputs, id: Id):
+        self._input = input
+        self._id = id
+
+    # TODO: Consider what to do if the outputs are not present right now. Should we
+    # allow a default value? Cause by default, it will throw a silent error.
+
+    def width(self):
+        return self._input[f".clientdata_output_{self._id}_width"]()
```

Comment:
Instead of implementing individual functions, it might be better to implement a `__getattr__` method. That would provide a place to handle outputs that don't exist yet.

Another useful thing with a `__getattr__` method is that it could return `reactive.Value` object itself, instead of only providing a way to invoke `()` on the object. Although... that could also be accomplished `@property` decorator -- and using that decorator on each individual item would also allow more specific typing, like `Value[bool]`.

Reply:
I don't like the `__getattr__` due to the inability to type or document each value (right?). The `@property` could work though. I don't love that `reactive.Value` exposes `.set()` that doesn't actually work, but the other methods on `Value` could be helpful.